src/app/components/authorized/add-new-post/add-new-post.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component } from '@angular/core';
+import { GatewayService } from 'src/app/services/gateway.service';
+import { Subreddit } from 'src/types/subreddits';
+import { NewPost } from 'src/types/posts';
+import { Router } from '@angular/router';
+import { Error } from 'src/types/error';
+
+@Component({
+    selector: 'app-add-new-post',
+    templateUrl: './add-new-post.component.html',
+    styleUrls: ['./add-new-post.component.scss'],
+})
+export class AddNewPostComponent {
+    newPost: NewPost = {
+        title: '',
+        subreddit: 'Select',
+        post_type: 'text',
+        commentCount: 0,
+        posted_url: '',
+        description: '',
+    };
+
+    namesOfSubreddits: string[];
+    urlExists: boolean;
+    errorMessage$?: Error;
+    allSubreddits$: Subreddit[];
+
+    constructor(private _gatewayService: GatewayService, private _router: Router) {
+        this._gatewayService.getAllSubreddits().subscribe((succes) => {
+            this.allSubreddits$ = succes;
+            this.getNamesOfSubreddits();
+            this.namesOfSubreddits.unshift('Select');
+        });
+    }
+
+    getNamesOfSubreddits(): string[] {
+        return (this.namesOfSubreddits = this.allSubreddits$.map((name) => name.name));
```

Comment:
I'm not sure how to feel about this. Although it is valid javascript and properly returns a value, it also sets a value. Which is not really a good pattern and leads to unclean code. This should be updated to better reflect what is happening.